.github/scripts/apidoc.sh

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
+#
+# Process (API) docs after a successful build on Travis (via ../.travis.yml).
+#
+# Updated/changed documentation for "master" is pushed to gh-pages.
+# In case of pull requests or other branches, it will get added to a separate branch.
+# In case of a pull request, a compare view comment will be posted.
+#
+# NOTE: stdout/stderr might/should be discarded to not leak sensitive information.
+
+echo "Post-processing (API) documentation."
+echo "PR Number: $PR_NUMBER"
+echo "GITHUB_HEAD_REF: $GITHUB_HEAD_REF"
+
+if [ -z "$APIDOC_TOKEN" ]; then
+    echo "No APIDOC_TOKEN available. Skipping."
+    exit
+fi
+
+# NOTE: DO NOT USE "set -x", or anything else that would reveal APIDOC_TOKEN!
+# GitHub Actions does filter secrets, but extra caution won't hurt
+set -e
+set +x
+
+# Display exit code in case of failure (probably due to 'set -x').
+trap '[ "$?" = 0 ] || echo "EXIT CODE: $?"' EXIT
+
+REPO_APIDOC="https://${APIDOC_TOKEN}@github.com/awesomeWM/apidoc"
+REPO_DIR="$PWD"
+
+export GIT_AUTHOR_NAME="awesome-robot on GitHub Actions"
+export GIT_AUTHOR_EMAIL="awesome-robot@users.noreply.github.com"
+export GIT_COMMITTER_NAME="$GIT_AUTHOR_NAME"
+export GIT_COMMITTER_EMAIL="$GIT_AUTHOR_EMAIL"
+
+git clone --depth 1 --branch gh-pages "$REPO_APIDOC" build/apidoc \
+    2>&1 | sed "s/$APIDOC_TOKEN/APIDOC_TOKEN/g"
+cd build/apidoc
+
+# This will re-use already existing branches (updated PR).
+if [ "$PR_NUMBER" != false ]; then
+    BRANCH="pr-$PR_NUMBER"
+elif [ "$GITHUB_HEAD_REF" != master ]; then
+    # Use merge-base of master in branch name, to keep different branches with
+    # the same name apart.
+    # shellcheck disable=SC2015
+    BRANCH="$GITHUB_HEAD_REF-$(cd "$REPO_DIR" \
+        && git fetch --unshallow origin master \
+        && git rev-parse --short "$(git merge-base HEAD FETCH_HEAD || true)" || true)"
```

Comment:
I might not understand this correctly.
This `if` part (`"$GITHUB_HEAD_REF" != master`) is when a commit is merged (git push), how can it have two branches with the same name ?
Even if it is possible, the merge base for all such branches would probably be `origin/master` so I'm not sure whether the issue is fixed in this code